connexion/operations/openapi.py

```diff
@@ -302,25 +318,11 @@ def _get_body_argument(self, body, arguments, has_kwargs, sanitize):
         # see: https://github.com/OAI/OpenAPI-Specification/blame/3.0.2/versions/3.0.2.md#L2305
         additional_props = self.body_schema.get("additionalProperties", True)
 
-        if body is None:
-            body = deepcopy(default_body)
-
-        # if the body isn't even an object, then none of the concerns below matter
-        if self.body_schema.get("type") != "object":
-            if x_body_name in arguments or has_kwargs:
-                return {x_body_name: body}
-            return {}
-
-        # supply the initial defaults and convert all values to the proper types by schema
         body_arg = deepcopy(default_body)
         body_arg.update(body or {})
 
-        res = {}
         if body_props or additional_props:
-            res = self._get_typed_body_values(body_arg, body_props, additional_props)
-
-        if x_body_name in arguments or has_kwargs:
-            return {x_body_name: res}
+            return self._get_typed_body_values(body_arg, body_props, additional_props)
```

Comment:
Why do we need this for form types and not json?

Reply:
Because for json, strings are already converted to appropriate python types via `json.loads`. But there's no such conversion for forms, [only string splitting of arrays](https://github.com/spec-first/connexion/blob/84e33e5897a6a43f2a06ee5193ea958386252ce2/connexion/decorators/uri_parsing.py#L181). So I have left forms to be processed the old way and focused on fixing json, which is simpler to fix and more important (`oneOf` is rare in form data).

In my opinion, proper solution would perform type conversions somehow via jsonschema validators, at one place, close to the start of the request handling pipeline. Currently, types are converted inconsistently [here](https://github.com/spec-first/connexion/blob/84e33e5897a6a43f2a06ee5193ea958386252ce2/connexion/decorators/validation.py#L59) and [there](https://github.com/spec-first/connexion/blob/84e33e5897a6a43f2a06ee5193ea958386252ce2/connexion/decorators/uri_parsing.py#L179) and [there](https://github.com/spec-first/connexion/blob/84e33e5897a6a43f2a06ee5193ea958386252ce2/connexion/operations/openapi.py#L326) and [there](https://github.com/spec-first/connexion/blob/84e33e5897a6a43f2a06ee5193ea958386252ce2/connexion/operations/swagger2.py#L309). But please, this was intended to be a quick simple bugfix MR, let's not dive into big refactorings here.